### PR TITLE
Pension fund statistics

### PR DESCRIPTION
**Pull request by ErkoRisthein**

### CODE REVIEW

File: src/main/java/ee/tuleva/onboarding/fund/FundService.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  }
+
+  private PensionFundStatistics statistics(Fund fund) {
+    return pensionFundStatisticsService.getCachedStatistics().stream()
+      .filter(statistic -> Objects.equals(statistic.getIsin(), fund.getIsin()))
+      .findFirst()
+      .orElse(PensionFundStatistics.NULL);
+  }
+
+  private Iterable<Fund> fundsBy(Optional<String> fundManagerName) {
```
Comment (JordanValdma) [inline src/main/java/ee/tuleva/onboarding/fund/FundService.java]: I'd probably separate these methods

File: src/main/java/ee/tuleva/onboarding/fund/FundService.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class FundService {
+
+  private final FundRepository fundRepository;
+  private final PensionFundStatisticsService pensionFundStatisticsService;
+
+  public List<FundResponse> getFunds(Optional<String> fundManagerName) {
```
Comment (JordanValdma) [inline src/main/java/ee/tuleva/onboarding/fund/FundService.java]: I'd probably separate these methods

File: src/main/java/ee/tuleva/onboarding/fund/statistics/PensionFundStatisticsResponse.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+import javax.xml.bind.annotation.XmlAccessType;
+import javax.xml.bind.annotation.XmlAccessorType;
+import javax.xml.bind.annotation.XmlElement;
+import javax.xml.bind.annotation.XmlRootElement;
+import java.util.List;
+
+@Data
+@XmlRootElement(name = "RESPONSE", namespace = "http://corporate.epis.ee/producer/")
+@XmlAccessorType(XmlAccessType.FIELD)
+public class PensionFundStatisticsResponse {
```
Comment (JordanValdma) [inline src/main/java/ee/tuleva/onboarding/fund/statistics/PensionFundStatisticsResponse.java]: Perhaps rename to DTO?

Comment (ErkoRisthein) [inline src/main/java/ee/tuleva/onboarding/fund/statistics/PensionFundStatisticsResponse.java]: I thought about it, but then saw that we use a lot of `*Request` and `*Response` class names, so I ended up following the "convention".

**Pull request merged**